engine/runtime/src/main/java/org/enso/interpreter/runtime/warning/Warning.java

```diff
@@ -74,116 +76,39 @@ public EnsoObject getReassignments() {
       autoRegister = false)
   @Builtin.Specialize
   public static WithWarnings attach(
-      EnsoContext ctx, WithWarnings value, Object warning, Object origin) {
-    return value.append(ctx, new Warning(warning, origin, ctx.nextSequenceId()));
-  }
-
-  @Builtin.Method(
-      name = "attach_with_stacktrace",
-      description = "Attaches the given warning to the value.",
-      autoRegister = false)
-  @Builtin.Specialize(fallback = true)
-  public static WithWarnings attach(EnsoContext ctx, Object value, Object warning, Object origin) {
-    return WithWarnings.wrap(ctx, value, new Warning(warning, origin, ctx.nextSequenceId()));
-  }
-
-  @Builtin.Method(
-      name = "get_all_vector",
-      description = "Gets all the warnings associated with the value.",
-      autoRegister = false)
-  @Builtin.Specialize
-  @CompilerDirectives.TruffleBoundary
-  public static EnsoObject getAll(
-      WithWarnings value, boolean shouldWrap, WarningsLibrary warningsLib) {
-    Warning[] warnings = value.getWarningsArray(warningsLib, shouldWrap);
-    sortArray(warnings);
-    return ArrayLikeHelpers.asVectorEnsoObjects(warnings);
-  }
-
-  @Builtin.Method(
-      name = "get_all_vector",
-      description = "Gets all the warnings associated with the value.",
-      autoRegister = false)
-  @Builtin.Specialize(fallback = true)
-  public static EnsoObject getAll(Object value, boolean shouldWrap, WarningsLibrary warningsLib) {
-    if (warningsLib.hasWarnings(value)) {
-      try {
-        Warning[] warnings = warningsLib.getWarnings(value, null, shouldWrap);
-        sortArray(warnings);
-        return ArrayLikeHelpers.asVectorEnsoObjects(warnings);
-      } catch (UnsupportedMessageException e) {
-        throw EnsoContext.get(warningsLib).raiseAssertionPanic(warningsLib, null, e);
+      EnsoContext ctx,
+      Object value,
+      Object warning,
+      Object origin,
+      @Cached AppendWarningNode appendWarningNode) {
+    var warn = new Warning(warning, origin, ctx.nextSequenceId());
+    return appendWarningNode.execute(null, value, warn);
+  }
+
+  public static Warning[] fromMapToArray(
+      EnsoHashMap map, ArrayLikeLengthNode lengthNode, ArrayLikeAtNode atNode) {
+    var vec = map.getCachedVectorRepresentation();
+    var vecLen = Math.toIntExact(lengthNode.executeLength(vec));
```

Comment:
Eh, will we crash the interpreter with `ArithmeticException` when the value doesn't fit into `int`?

Reply:
900b091fa1ccbb5e6b8a479bb8403e1f9674e5d0 just casts the `long` to `int`. It is very unlikely that there is a long that is greater than Integer.MAX anyway.